apps/bfd-server/bfd-server-launcher/src/main/java/gov/cms/bfd/server/launcher/DataServerLauncherApp.java

```diff
@@ -351,9 +363,30 @@ public void log(Request request, Response response) {
          * the underlying Jetty classes in the response that are in classes that are not loaded in the war file so not
          * accessible to the filter.
          */
-        BfdMDC.put(
-            HTTP_ACCESS_RESPONSE_OUTPUT_SIZE_IN_BYTES,
-            String.valueOf(response.getHttpOutput().getWritten()));
+        Long outputSizeInBytes = response.getHttpOutput().getWritten();
+        BfdMDC.put(HTTP_ACCESS_RESPONSE_OUTPUT_SIZE_IN_BYTES, String.valueOf(outputSizeInBytes));
+
+        // Record the response duration.
+        Long requestStartMilliseconds =
+            (Long) request.getAttribute(HTTP_ACCESS_RESPONSE_START_MILLISECONDS);
+        if (requestStartMilliseconds != null) {
+          Long responseDurationInMilliseconds =
+              System.currentTimeMillis() - requestStartMilliseconds;
+          BfdMDC.put(
+              BfdMDC.computeMDCKey(HTTP_ACCESS_RESPONSE_DURATION_MILLISECONDS),
+              Long.toString(responseDurationInMilliseconds));
+
+          if (outputSizeInBytes != 0 && responseDurationInMilliseconds != 0) {
+
+            Long responseDurationPerKB =
+                ((1024 * responseDurationInMilliseconds) / outputSizeInBytes);
+            BfdMDC.put(HTTP_ACCESS_RESPONSE_DURATION_PER_KB, String.valueOf(responseDurationPerKB));
+          } else {
+            BfdMDC.put(HTTP_ACCESS_RESPONSE_DURATION_PER_KB, null);
+          }
+        } else {
+          BfdMDC.put(HTTP_ACCESS_RESPONSE_DURATION_PER_KB, null);
+        }
 
         /*
          * Write to the access.json. The message here isn't actually the payload; the MDC context that will get
```

Comment:
going to leave a comment here for something that is goign to be a new requirement -- updating the terraform for bfd insights to add the new column. we'll work with @andrew-kramer on that since he is still getting the PR ready for that terraform to be present at all.

Reply:
we'll not hold for this after all